scripts/test_network_reachable.py

```diff
@@ -0,0 +1,163 @@
+    elif archend=='mipsel':
+        return 'qemu-system-mipsel'
+def get_scratch(iid):
+    return path.join(SCRATCH_DIR, '%d'%iid)
+def get_fs(iid):
+    return path.join(get_scratch(iid),'image.raw')
+def closeIp(ip):
+    tups = [int(x) for x in ip.split(".")]
+    if tups[3] != 1:
+        tups[3] -= 1
+    else:
+        tups[3] = 2
+    return ".".join([str(x) for x in tups])
+
+
+
+if len(sys.argv)<2:
+    print("usage: \n\
+            %s {IID}"%(sys.argv[0]))
+    sys.exit()
+IID = int(sys.argv[1])
+cur.execute("SELECT arch FROM image WHERE id=%s"%IID)
+ARCHEND = cur.fetchone()[0]
+cur.execute("SELECT guest_ip FROM image WHERE id=%d"%IID)
+GUESTIP = cur.fetchone()[0]
+NETDEVIP=closeIp(GUESTIP)
+TAPDEV='tap%d'%IID
+HOSTNETDEV=TAPDEV
+
+IMAGE=get_fs(IID)
+KERNEL=get_kernel(ARCHEND)
+QEMU=get_qemu(ARCHEND)
+QEMU_MACHINE=get_qemu_machine(ARCHEND)
+QEMU_ROOTFS=get_qemu_disk(ARCHEND)
+WORK_DIR=get_scratch(IID)
+
+print("Creating TAP device {TAPDEV}...".format(**gl(locals())))
+shell('sudo tunctl -t {TAPDEV} -u $USER'.format(**gl(locals())))
+print("Bringing up TAP device...")
+shell('sudo ifconfig {HOSTNETDEV} {NETDEVIP}/24 up'.format(**gl(locals())))
+print("Adding route to {GUESTIP}...".format(**gl(locals())))
+shell('sudo route add -host {GUESTIP} gw {GUESTIP} {HOSTNETDEV}'.format(**gl(locals())))
+
+def ifaceNo(dev):
+    g = re.match(r"[^0-9]+([0-9]+)", dev)
+    return int(g.group(1))
+def qemuNetworkConfig(dev):
+    result = ""
+    for i in range(0, 4):
+        if i == ifaceNo(dev):
+            result += "-net nic,vlan=%i -net tap,vlan=%i,ifname={TAPDEV},script=no ".format(**gl(locals())) \
+                    % (i, i)
+        else:
+            result += "-net nic,vlan=%i -net socket,vlan=%i,listen=:200%i "% (i, i, i)
+    return result
+
+print("Starting emulation of firmware... ")
+if ARCHEND=='mipsel' or ARCHEND=='mipseb':
+    qemuEnvVars=""
+    qemuDisk = "-drive if=ide,format=raw,file={IMAGE}".format(**locals())
+elif ARCHEND=='armel':
+    qemuEnvVars = "QEMU_AUDIO_DRV=none "
+    qemuDisk = "-drive if=none,file={IMAGE},format=raw,id=rootfs -device virtio-blk-device,drive=rootfs".format(**locals())
+cur.execute("SELECT netdev FROM image WHERE id=%d"%IID)
+netdev=cur.fetchone()[0]
+qemuNetwork=qemuNetworkConfig(netdev)
+
+
+shell('''{qemuEnvVars} {QEMU} -m 256 -M {QEMU_MACHINE} -kernel {KERNEL} \
+        {qemuDisk} -append "root={QEMU_ROOTFS} console=ttyS0 \
+        nandsim.parts=64,64,64,64,64,64,64,64,64,64 rdinit=/preInit.sh rw debug ignore_loglevel \
+        print-fatal-signals=1 user_debug=31 firmadyne.syscall=8" \
+        -serial file:{WORK_DIR}/qemu.final.serial.log \
+        -display none \
+        -daemonize \
+        {qemuNetwork}'''.format(**gl(locals())))
+
+# the good sleep time depends on the time whether eth1 appear in the serial.log
+time.sleep(10)
+network_reachable=False
+try:
+    with request.urlopen('http://{GUESTIP}/'.format(**gl(locals())), timeout=10) as fin:
```

Comment:
But you need to perform ping manually, right? Otherwise, you don't know the good timing when to call the ping and determine the whether the network is reachable.